include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf_coupled_solver.cpp

```diff
@@ -225,22 +226,48 @@ TimeIntBDFCoupled<dim, Number>::do_timestep_solve()
     ((this->time_step_number - 1) % this->param.update_preconditioner_coupled_every_time_steps ==
      0);
 
-  if(this->param.nonlinear_problem_has_to_be_solved())
+  // calculate Sum_i (alpha_i/dt * u_i) and store
+  VectorType sum_alphai_ui(solution[0].block(0));
+  sum_alphai_ui.equ(this->bdf.get_alpha(0) / this->get_time_step_size(), solution[0].block(0));
+  for(unsigned int i = 1; i < solution.size(); ++i)
   {
-    VectorType sum_alphai_ui(solution[0].block(0));
+    sum_alphai_ui.add(this->bdf.get_alpha(i) / this->get_time_step_size(), solution[i].block(0));
+  }
 
-    // calculate Sum_i (alpha_i/dt * u_i)
-    sum_alphai_ui.equ(this->bdf.get_alpha(0) / this->get_time_step_size(), solution[0].block(0));
-    for(unsigned int i = 1; i < solution.size(); ++i)
+  // Update the convective term when using an ALE formulation
+  if(this->param.convective_problem() and
+     this->param.treatment_of_convective_term == TreatmentOfConvectiveTerm::Explicit)
+  {
+    if(this->param.ale_formulation)
     {
-      sum_alphai_ui.add(this->bdf.get_alpha(i) / this->get_time_step_size(), solution[i].block(0));
+      // evaluate convective term for all previous times since the mesh has been updated
+      for(unsigned int i = 0; i < this->vec_convective_term.size(); ++i)
+      {
+        // in a general setting, we only know the boundary conditions at time t_{n+1}
+        pde_operator->evaluate_convective_term(this->vec_convective_term[i],
+                                               solution[i].block(0),
+                                               this->get_next_time());
+      }
     }
+  }
 
+  if(this->param.nonlinear_problem_has_to_be_solved())
+  {
     VectorType rhs(sum_alphai_ui);
     pde_operator->apply_mass_operator(rhs, sum_alphai_ui);
     if(this->param.right_hand_side)
       pde_operator->evaluate_add_body_force_term(rhs, this->get_next_time());
 
+    // Add the convective term to the right-hand side of the equations
+    // if the convective term is treated explicitly (additive decomposition):
+    // evaluate convective term and add extrapolation of convective term to the rhs (-> minus sign!)
```

Comment:
"evaluate convective term and" needs to be removed here in order to avoid irritations.

Reply:
adapted both :)